In [146]:
import pandas as pd

In [147]:
import numpy as np
import os
cwd = os.getcwd()
os.chdir('/Users/shantanupatil/Desktop/Dataset/aspiring_minds/datachallenge_cods2016/datachallenge_cods2016/')

In [148]:
os.listdir('.')

['data_description_doc.xlsx',
 'Dataset Instructions.pdf',
 'results.xlsx',
 'test.xlsx',
 'train.xlsx']

In [149]:
file ='train.xlsx'
xl = pd.ExcelFile(file)

In [150]:
print(xl.sheet_names)

['Sheet1']


In [151]:
df = xl.parse('Sheet1')

In [152]:
df.size

151924

In [153]:
df = df.drop(df.columns[[2,3,5]], axis=1) 

In [154]:
df.head()

ID   Salary               Designation Gender        DOB  \
train  203097   420000   senior quality engineer      f 1990-02-19   
train  579905   500000         assistant manager      m 1989-10-04   
train  810601   325000          systems engineer      f 1992-08-03   
train  267447  1100000  senior software engineer      m 1989-12-05   
train  343523   200000                       get      m 1991-02-27   

       10percentage                         10board  12graduation  \
train          84.3  board ofsecondary education,ap          2007   
train          85.4                            cbse          2007   
train          85.0                            cbse          2010   
train          85.6                            cbse          2007   
train          78.0                            cbse          2008   

       12percentage                             12board  \
train          95.8  board of intermediate education,ap   
train          85.0                                cbse   
train          68.2                                cbse   
train          83.6                                cbse   
train          76.8                                cbse   

               ...            ComputerScience  MechanicalEngg ElectricalEngg  \
train          ...                         -1              -1             -1   
train          ...                         -1              -1             -1   
train          ...                         -1              -1             -1   
train          ...                         -1              -1             -1   
train          ...                         -1              -1             -1   

      TelecomEngg  CivilEngg  conscientiousness  agreeableness extraversion  \
train          -1         -1             0.9737         0.8128       0.5269   
train          -1         -1            -0.7335         0.3789       1.2396   
train          -1         -1             0.2718         1.7109       0.1637   
train          -1         -1             0.0464         0.3448      -0.3440   
train          -1         -1            -0.8810        -0.2793      -1.0697   

       nueroticism  openess_to_experience  
train      1.35490                -0.4455  
train     -0.10760                 0.8637  
train     -0.86820                 0.6721  
train     -0.40780                -0.9194  
train      0.09163                -0.1295  

[5 rows x 35 columns]

In [155]:
df = df.drop(df.columns[4], axis=1) 

In [156]:
list(df)

['ID',
 'Salary',
 'Designation',
 'Gender',
 '10percentage',
 '10board',
 '12graduation',
 '12percentage',
 '12board',
 'CollegeID',
 'CollegeTier',
 'Degree',
 'Specialization',
 'collegeGPA',
 'CollegeCityID',
 'CollegeCityTier',
 'CollegeState',
 'GraduationYear',
 'English',
 'Logical',
 'Quant',
 'Domain',
 'ComputerProgramming',
 'ElectronicsAndSemicon',
 'ComputerScience',
 'MechanicalEngg',
 'ElectricalEngg',
 'TelecomEngg',
 'CivilEngg',
 'conscientiousness',
 'agreeableness',
 'extraversion',
 'nueroticism',
 'openess_to_experience']

In [157]:
df = df.drop(df.columns[[9,14,16,17,22,23,24,25,26,27,28]], axis = 1)

In [158]:
list(df)

['ID',
 'Salary',
 'Designation',
 'Gender',
 '10percentage',
 '10board',
 '12graduation',
 '12percentage',
 '12board',
 'CollegeTier',
 'Degree',
 'Specialization',
 'collegeGPA',
 'CollegeCityTier',
 'English',
 'Logical',
 'Quant',
 'Domain',
 'conscientiousness',
 'agreeableness',
 'extraversion',
 'nueroticism',
 'openess_to_experience']

In [159]:
df.groupby('Degree').size()

Degree
B.Tech/B.E.      3700
M.Sc. (Tech.)       2
M.Tech./M.E.       53
MCA               243
dtype: int64

In [160]:
df

ID   Salary                   Designation Gender  10percentage  \
train   203097   420000       senior quality engineer      f         84.30   
train   579905   500000             assistant manager      m         85.40   
train   810601   325000              systems engineer      f         85.00   
train   267447  1100000      senior software engineer      m         85.60   
train   343523   200000                           get      m         78.00   
train  1027655   300000               system engineer      m         89.92   
train   947847   300000        java software engineer      m         86.08   
train   912934   400000           mechanical engineer      m         92.00   
train   552574   600000           electrical engineer      m         90.00   
train  1203363   230000              project engineer      m         77.00   
train    87291   600000          senior php developer      m         88.60   
train   190393   450000       senior systems engineer      m         81.00   
train   606655   270000    quality assurance engineer      m         83.60   
train   431800   200000                    qa analyst      m         60.80   
train  1279958   300000        java software engineer      m         81.20   
train   536517   350000              network engineer      f         69.54   
train  1087625   325000  product development engineer      m         85.80   
train   874596   250000  associate software developer      m         60.80   
train   711342   120000           data entry operator      m         65.00   
train   466888   325000             software engineer      f         79.00   
train   140069   320000             software engineer      f         91.20   
train   339689   200000             software engineer      f         75.67   
train   995191   190000                     developer      m         92.50   
train   990077   180000   electrical project engineer      m         70.00   
train   963123   335000            programmer analyst      m         88.00   
train   350211   435000               systems analyst      f         86.80   
train    65096   345000      senior software engineer      m         90.88   
train   810458   300000                           ase      m         76.80   
train   918981   145000    telecommunication engineer      m         84.00   
train   472956   220000        java software engineer      m         64.00   
...        ...      ...                           ...    ...           ...   
train   469012   360000             software engineer      f         81.00   
train  1233826   330000            technical engineer      m         76.00   
train  1092890   310000             software engineer      m         82.80   
train   143926   500000       senior systems engineer      f         81.00   
train  1044603   180000             help desk analyst      m         68.20   
train   755067   335000            programmer analyst      m         66.85   
train   641821   180000              quality engineer      m         70.80   
train  1240207   300000                game developer      m         86.00   
train  1006997   300000     graduate engineer trainee      f         64.20   
train   217036   415000             software engineer      m         62.00   
train  1274658   320000             software engineer      f         89.70   
train   212055   550000             software engineer      m         69.16   
train   197796   150000                           get      m         84.00   
train  1077872   220000             software engineer      m         53.40   
train   553436   160000        java software engineer      m         83.00   
train  1076441   315000          system administrator      m         88.10   
train   305041   480000             software engineer      f         89.80   
train   358393   430000               system engineer      f         90.40   
train    51897   225000        software test engineer      f         84.93   
train   439787   280000           

In [161]:
df2 = df['Salary'].rank(axis=0, method='average', numeric_only=None, na_option='keep', ascending=False, pct=False)

In [162]:
dfnew = pd.concat([df2, df], axis=1)

In [163]:
df.size


91954

In [164]:

dfsucess = df['Salary']>=800000

In [165]:
dfsucess = df[dfsucess]

In [169]:
dfsucess[["Salary","Gender","10percentage","12percentage","CollegeTier","collegeGPA",'English',
 'Logical',
 'Quant',
 'Domain',
 'conscientiousness',
 'agreeableness',
 'extraversion',
 'nueroticism',
 'openess_to_experience']].describe()

Salary  10percentage  12percentage  CollegeTier  collegeGPA  \
count  6.300000e+01     63.000000     63.000000    63.000000   63.000000   
mean   1.386984e+06     79.580476     75.601429     1.714286   73.283333   
std    7.602231e+05     10.554252     11.915717     0.455383   12.234770   
min    8.000000e+05     53.000000     50.000000     1.000000    8.890000   
25%    8.550000e+05     72.750000     67.600000     1.000000   67.400000   
50%    1.100000e+06     82.400000     76.670000     2.000000   72.600000   
75%    1.772500e+06     87.800000     86.625000     2.000000   79.950000   
max    4.000000e+06     97.760000     96.330000     2.000000   96.900000   

          English     Logical       Quant     Domain  conscientiousness  \
count   63.000000   63.000000   63.000000  63.000000          63.000000   
mean   526.253968  524.761905  573.841270   0.651155          -0.200500   
std    115.045008  104.083669  149.841228   0.436343           1.075514   
min    305.000000  315.000000  235.000000  -1.000000          -3.353900   
25%    449.500000  445.000000  485.000000   0.537125          -0.881000   
50%    520.000000  525.000000  595.000000   0.765674          -0.159000   
75%    630.000000  610.000000  694.500000   0.916870           0.415500   
max    735.000000  695.000000  840.000000   0.994051           1.592000   

       agreeableness  extraversion  nueroticism  openess_to_experience  
count      63.000000     63.000000    63.000000              63.000000  
mean        0.321483      0.235922    -0.469067              -0.159310  
std         0.855994      0.899548     0.849850               0.932729  
min        -1.785600     -1.650200    -2.287900              -2.815200  
25%        -0.084200     -0.344000    -0.877800              -0.715350  
50%         0.500800      0.381700    -0.538200              -0.094300  
75%         0.968800      0.874650    -0.055200               0.502400  
max         1.748800      2.535400     1.707430               1.247000

sucessful people = earning above 8 lack

Thus by looking at the data we can say that 

    1.scoreing low in CollegeTier 1 will have more probablity of sucess than scoreing same in CollegeTier 2

    2.Avg percetage of people earning more than 8 lakh in 10th = 79.5 and in 12th = 75 

    3.Avg gpa was 73.23

    4.Minimum percetage of people earning more than 8 lakh in 10th = 53 and in 12th = 50 but they improved there college GPA TO 8.89

    5.from psycological stand point average succesful people have agreeableness , extraversion  qualities.
    
    6.from psycological stand point most succesful people have  conscientiousness ,agreeableness, extraversion,	          openess_to_experience qualities and have more score than average people

In [167]:
dfsucess.head()

ID   Salary               Designation Gender  10percentage  \
train   267447  1100000  senior software engineer      m         85.60   
train   361583   800000         software engineer      m         93.44   
train  1250429  1500000     application developer      m         79.00   
train   312164  1200000          engineer trainee      m         59.80   
train   323894  1860000            field engineer      f         89.70   

                     10board  12graduation  12percentage  \
train                   cbse          2007         83.60   
train  karnataka state board          2008         90.00   
train            state board          2009         87.90   
train                   icse          2006         60.25   
train                      0          2007         89.60   

                     12board  CollegeTier          ...            \
train                   cbse            1          ...             
train  karnataka state board            2          ...             
train            state board            2          ...             
train                    isc            2          ...             
train                      0            1          ...             

      CollegeCityTier English  Logical  Quant    Domain  conscientiousness  \
train               1     635      585    625  0.974396             0.0464   
train               0     515      475    635  0.942117            -0.4173   
train               1     315      435    355  0.356536             0.4155   
train               0     575      595    405  0.021066             0.2009   
train               0     625      555    625  0.910770            -1.3447   

       agreeableness  extraversion  nueroticism  openess_to_experience  
train         0.3448       -0.3440     -0.40780                -0.9194  
train         0.9688       -0.1988     -0.29020                 0.3049  
train         0.5454        0.9322     -0.61470                 0.8637  
train         1.1248        1.1074     -1.11280                 0.9763  
train         0.6178       -1.4688      0.88483                 0.6603  

[5 rows x 23 columns]